src/finch/tensor.py

```diff
@@ -334,3 +349,62 @@ def permute_dims(x: Tensor, axes: tuple[int, ...]):
 
 def _is_scipy_sparse_obj(x):
     return hasattr(x, "__module__") and x.__module__.startswith("scipy.sparse")
+
+
+def _slice_plus_one(s: slice, size: int) -> range:
+    new_start = s.start + 1 if s.start is not None else 1
+    new_step = s.step if s.step is not None else 1
+    new_stop = s.stop if s.stop is not None else size
```

Comment:
We might want to sanitize all the integers/indices, as they can be negative and there are many special cases.

Reply:
Oh yes really good point! We need to sanitize the negative indices!

Reply:
Sure! I implemented support (with test cases) for negative indices and slices with negative `start`, `stop` or `step` values. There's a few `if` statements to match Python `slice` to Julia `range`.